app/code/core/Mage/Catalog/sql/catalog_setup/upgrade-1.6.0.0.19.1.6-1.6.0.0.19.1.7.php

```diff
@@ -0,0 +1,38 @@
+<?php
+/**
+ * Magento
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@magento.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade Magento to newer
+ * versions in the future. If you wish to customize Magento for your
+ * needs please refer to http://www.magento.com for more information.
+ *
+ * @category    Mage
+ * @package     Mage_Catalog
+ * @copyright  Copyright (c) 2006-2020 Magento, Inc. (http://www.magento.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+/** @var Mage_Catalog_Model_Resource_Setup $installer */
+$installer = $this;
+$connection = $installer->getConnection();
+$connection->addColumn($installer->getTable('core_url_rewrite'), 'url_type', 'varchar(50) NULL AFTER `product_id`');
```

Comment:
I think it's missing an index here. Probably several indexes.

Reply:
or better like colinmollenhour say's: change `varchar` to  `enum`?

Reply:
Either way, it should still have an index.

Reply:
If you leave off the `AFTER product_id` part MySQL 8.0 will be able to add the column without copying the entire table  which can reduce downtime while upgrading.